woocommerce/woocommerce
# Fix TT2 button styles

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces the following changes:
- Wraps TT2's main button styles under the existence of the `woocommerce-block-theme-has-button-styles` class in the body.
- Replaces the `word-wrap: break-word` rule with the `white-space: normal`, to resolve issues with buttons in a table. See example: https://d.pr/i/xrjT9A
- Adds default color values in all CSS variables for buttons.
- Fixes button colors to be aligned with the colors described in the _theme.json_ file.

Here's an example of the end result in a single product view using the global _Blue_ style:
![screencapture-twentytwentytwo-test-product-beanie-2023-04-05-17_42_36](https://user-images.githubusercontent.com/1105590/230116950-502e6892-0688-4690-b704-b923f51a958e.png)

Closes #37578



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:
 
**Testing for global styles:**

1. Create a site using the TT2 theme; or an existing one.
2. Change the global style to Blue and notice the button issues.
3. Apply this fix, and ensure that the buttons appear correctly.
